runtimes/dart-2.18/server.dart

```diff
@@ -10,56 +10,121 @@ import 'function_types.dart';
-      return shelf.Response(500, body: 'Unauthorized');
+
+    String bodyString = await req.readAsString();
+    dynamic body = bodyString;
+    String method = req.method;
+    Map<String, dynamic> headers = {};
+
+    for (MapEntry entry in req.headers.entries) {
+      String header = entry.key.toLowerCase();
+      if(!header.startsWith('x-open-runtimes-')) {
+        headers[header] = entry.value;
+      }
+    }
+
+    String contentType = req.headers['content-type'] ?? 'plain/text';
+    if(contentType.contains('application/json')) {
+      if(!bodyString.isEmpty) {
+        body = jsonDecode(bodyString);
+      } else {
+        body = {};
+      }
+    }
+
+    String hostHeader = req.headers['host'] ?? '';
+
+    String path = '/' + req.url.path;
+    String host = '';
+    String scheme = req.headers['x-forwarded-proto'] ?? 'http';
+    String defaultPort = scheme == 'https' ? '443' : '80';
+    int port = int.parse(defaultPort);
+    String queryString = req.url.query;
+    Map<String, String> query = {};
+
+    if(hostHeader.contains(':')) {
+      host = hostHeader.split(':')[0];
+      port = int.parse(hostHeader.split(':')[1]);
+    } else {
+      host = hostHeader;
+      port = int.parse(defaultPort);
+    }
+
+    for(final param in queryString.split("&")) {
+      final parts = param.split("=");
+
+      final key = parts[0];
+      final value = parts.sublist(1).join('=');
+
+      if(key != null && !key.isEmpty) {
+        query[key] = value;
+      }
+    }
+
+    String url = scheme + '://' + host;
+
+    if(port != int.parse(defaultPort)) {
+      url += ':' + port.toString();
     }
-    final headers = {
-      'content-type': 'application/json'
-    };
-    
-    try {
-      final bodystring = await req.readAsString();
-      final body = jsonDecode(bodystring);
-      final request = Request(
-        variables: body['variables'] ?? {},
-        headers: body['headers'] ?? {},
-        payload: body['payload'] ?? '',
-      );
 
-      final response = Response();
-      await runZonedGuarded(
+    url += path;
+
+    if(!queryString.isEmpty) {
+      url += '?' + queryString;
+    }
+
+    Request contextReq = new Request(bodyString: bodyString, body: body, headers: headers, method: method, url: url, path: path, port: port, scheme: scheme, host: host, queryString: queryString, query: query);
+    Response contextRes = new Response();
+    Context context = new Context(contextReq, contextRes);
+
+    String customstd = "";
+
+    dynamic output = null;
+    try {
+      await runZoned(
         () async {
-          await user_code.start(request, response);
+          output = await user_code.handle(context);
         },
-        (e, stackTrace) => print('$e $stackTrace'),
         zoneSpecification: ZoneSpecification(
           print: (Zone self, ZoneDelegate parent, Zone zone, String line) {
-            userLogs.add(line);
+            customstd += line;
           },
         ),
       );
```

Comment:
if we remove support for `console.log` in node, here as well we can remove support for `print` overwriting, so we can remove everything inside try and simplify it
```suggestion
          output = await user_code.handle(context);
```

Reply:
We are keeping console.log logic.

When you use natice way to log, we capture it. Instead of logging the message, we write a log saying "Please use context.log instead".

This way devs who use native log for first time (natural behaviour) will see exactly what they need to do to log properly.

Reply:
> This way devs who use native log for first time (natural behaviour) will see exactly what they need to do to log properly.

I don't quite agree with this approach. rewriting standard functionality has caused many problems in our current version of Appwrite. Sometimes, people do want to see the output in the docker logs.